packages/foam-cli/src/commands/janitor.ts

```diff
@@ -0,0 +1,73 @@
+import { Command, flags } from '@oclif/command'
+import { initializeNoteGraph, generateLinkReferences, generateHeading, getKebabCaseFileName } from 'foam-core';
+import * as path from 'path';
+import { applyTextEdit } from '../utils/apply-text-edit';
+import { writeFileToDisk } from '../utils/write-file-to-disk';
+import { renameFile } from '../utils/rename-file';
+
+
+export default class Janitor extends Command {
+  static description = 'Updates link references and heading across all the markdown files in the given workspaces';
+
+  static examples = [
+    `$ foam-cli janitor path-to-foam-workspace
+Sucessfully generated link references and heading!
+`,
+  ]
+
+  static flags = {
+    help: flags.help({ char: 'h' }),
+  }
+
+  static args = [{ name: 'workspacePath' }]
+
+  async run() {
+    const { args, flags } = this.parse(Janitor)
+
+    const { workspacePath = './' } = args;
+
+    const foamWorkspaceDir = path.join(__dirname, workspacePath);
+
+    const graph = await initializeNoteGraph(foamWorkspaceDir);
+
+    const notes = graph.getNotes();
+
+    const fileWritePromises = notes.map(note => {
+      // Generate Heading 
+      const heading = generateHeading(note);
+
+      const fileWithHeading = heading ? applyTextEdit(note.source, heading) : note.source;
```

Comment:
Usually variable reassignment is bad, but in this case I think it would improve readability to reassign to same instead of keep coming up with names.